### PR TITLE
[IOPAE-793] Frontend topic management on create/update service

### PR DESCRIPTION
**Pull request by giuseppedipinto**
Added **topic** management when **creating/editing** a service.
Service topic is an optional property that can be set in the create/update form.

#### List of Changes
- update backoffice `openapi` with topic data
- update Service **types**, **adapters** and all topic **related data**
- update **translations**
- update `MSW` mocks with topic data
- add new `AutocompleteController` with **MUI** `Autocomplete` component
- add **frontend logic** to manage service topic on create/update form
- minor adjustments

#### List of Changes (update)
- c90f589 fix MUI warning _(MUI: The value provided to Autocomplete is invalid. None of the options match with `""`.)_
- 35975fc update topic field layout: reduce width to half

#### Motivation and Context
Need to manage the topic of a service

#### How Has This Been Tested?
Unit tests
Local `dev` mode with `msw` mocks

#### Screenshots (if appropriate):

https://github.com/pagopa/io-services-cms/assets/118166285/b00f27f1-ee8c-40b2-bdf5-b9e44d4ce2cc

Updated layout:

https://github.com/pagopa/io-services-cms/assets/118166285/d176b9c8-9aed-4b5c-9aa7-53f3247d538b

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/backoffice/src/components/services/adapters.ts [hunk 1]
```diff
@@ -228,11 +227,11 @@ const getCtaUrlFromCtaString = (value: string) => {
  * or on future uses of such fields which may be undefined.
  * @param s service
  * @returns service without epmpty or undefined properties */
-const clearUndefinedEmptyProperties = (s: any): any => {
+const clearUndefinedNullEmptyProperties = (s: any): any => {
```
Comment (gunzip) [inline apps/backoffice/src/components/services/adapters.ts:230]: this is the only method in this repo that requires lodash. lodash provides the same features as fp-ts in a more imperative fashion. I strongly suggest to remove lodash in order to reduce the size of the bundle and rewrite this method. You can use plain javascript:

```ts
/** Remove temporary assistanceChannels[] metadata used for react-hook-form dynamic array fields. */
const removeAssistanceChannelsArray = (s: any) => {
  const { assistanceChannels, ...metadata } = s.metadata;
  return { ...s, metadata };
};

type DeepPartial<T> = {
  [P in keyof T]?: DeepPartial<T[P]>;
};

function clearUndefinedNullEmptyProperties<T extends object>(
  obj: T,
  filterOutValues = new Set([null, undefined, ""])
): DeepPartial<T> {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    if (filterOutValues.has(value)) {
      return acc;
    }

    if (typeof value === "object" && !Array.isArray(value)) {
      return { ...acc, [key]: clearUndefinedNullEmptyProperties(value) };
    }

    return { ...acc, [key]: value };
  }, {});
}

```

**Review by gunzip — CHANGES_REQUESTED** [review]

File: apps/backoffice/src/components/services/adapters.ts (hunk 1 above)
Comment (giuseppedipinto) [reply apps/backoffice/src/components/services/adapters.ts:230]: Hi @gunzip, I actually use lodash also in another component _(for object deep comparison)_.
In general I consider `lodash` a very valid _(and widely used)_ library. 
Furthermore, IMO, it is convenient and powerful as it expresses concepts in a minimal way.
However, I accept your objection and I have opened an internal technical debt [task](https://pagopa.atlassian.net/browse/IOPAE-902) to evaluate the problem globally on the repo.

Comment (gunzip) [reply apps/backoffice/src/components/services/adapters.ts:230]: Can you share a link with the code where lodash is used?

Comment (giuseppedipinto) [reply apps/backoffice/src/components/services/adapters.ts:230]: https://github.com/pagopa/io-services-cms/blob/b1e6dec1059b7b75b02c96c667e37ade2ac2b91e/apps/backoffice/src/components/create-update-process/create-update-process.tsx#L103

Comment (gunzip) [reply apps/backoffice/src/components/services/adapters.ts:230]: You can use something alike

```ts
export const isDeepEqual = (a: any, b: any): boolean => {
  if (a === b) {
    return true;
  }

  if (typeof a !== typeof b) {
    return false;
  }

  if (typeof a !== "object" || a === null || b === null) {
    return false;
  }

  if (Object.keys(a).length !== Object.keys(b).length) {
    return false;
  }

  if (Array.isArray(a) !== Array.isArray(b)) {
    return false;
  }

  if (Array.isArray(a)) {
    return a.every((item, index) => isDeepEqual(item, b[index]));
  }

  return Object.keys(a).every(key => isDeepEqual(a[key], b[key]));
};

```

or

```ts
export const isDeepEqual = (a: any, b: any): boolean => {
  return JSON.stringify(a) === JSON.stringify(b);
};

```

Comment (giuseppedipinto) [reply apps/backoffice/src/components/services/adapters.ts:230]: Sorry but it's not the same.
json.stringify assumes that the strings are ordered in the same way _(too ugly to look at! )_ while in the other case it doesn't seem acceptable to me to write about 20 lines of codes _(which also have to be tested)_ to do a job that I can achieve with one line...it seems more like a trial of intentions than a real need to eliminate a library.
Anyway we will evaluate this possibility in the technical debt task I reported above, but I would like to let you know that I do not agree with the approach

Comment (gunzip) [reply apps/backoffice/src/components/services/adapters.ts:230]: In which case 20 lines of code are worse than serving an entire library just to use 2 methods? Furthermore, consider in your evalutation that 80% of lodash features are covered by fp-ts which is a library you're already using.

Comment (giuseppedipinto) [reply apps/backoffice/src/components/services/adapters.ts:230]: I will evaluate it in internal technical debt [task](https://pagopa.atlassian.net/browse/IOPAE-902).
Thank you so much for the valuable advice and food for thought, as always 🙏😉

**Review by gunzip — APPROVED** [review]

**Review by StefanoDoc — APPROVED** [review]

**Pull request merged**